pkg/virt-controller/watch/vmi.go

```diff
@@ -2447,3 +2449,42 @@ func (c *VMIController) syncMemoryHotplug(vmi *virtv1.VirtualMachineInstance) {
 		vmi.Labels[virtv1.MemoryHotplugOverheadRatioLabel] = *overheadRatio
 	}
 }
+
+func (c *VMIController) aggregateDataVolumesConditions(vmiCopy *virtv1.VirtualMachineInstance, dvs []*cdiv1.DataVolume) {
+	if len(dvs) == 0 {
+		return
+	}
+
+	dvsReadyCondition := virtv1.VirtualMachineInstanceCondition{
+		Status:  k8sv1.ConditionTrue,
+		Type:    virtv1.VirtualMachineInstanceDataVolumesReady,
+		Reason:  virtv1.VirtualMachineInstanceReasonAllDVsReady,
+		Message: "All of the VMI's DVs are bound and not running",
+	}
+
+	for _, dv := range dvs {
+		cstatus := getReadyConditionStatus(dv.Status.Conditions)
+		if cstatus != k8sv1.ConditionTrue {
+			dvsReadyCondition.Reason = virtv1.VirtualMachineInstanceReasonNotAllDVsReady
+			if cstatus == k8sv1.ConditionFalse {
+				dvsReadyCondition.Message = "Not all of the VMI's DVs are ready"
+				dvsReadyCondition.Status = cstatus
+			} else if cstatus == k8sv1.ConditionUnknown && dvsReadyCondition.Status == k8sv1.ConditionTrue {
```

Comment:
Why do you update dvsReady with `Unknown` only if it previously was `True`?

Reply:
Because if we already know that it's `False` due to another DV it's not completely `Unknown`